walk-in-interview/src/test/java/com/google/job/data/JobsDatabaseTest.java

```diff
@@ -245,6 +246,182 @@ public void fetchJob_normalInput_success() throws ExecutionException, Interrupte
+                .setJobDuration(jobDuration)
+                .setLocation(location1)
+                .setJobPay(jobPayment1)
+                .build();
+
+        Job job2 = Job.newBuilder()
+                .setJobStatus(jobStatus)
+                .setJobTitle(jobName)
+                .setJobDescription(jobDescription)
+                .setRequirements(requirements)
+                .setPostExpiry(postExpiry)
+                .setJobDuration(jobDuration)
+                .setLocation(location2)
+                .setJobPay(jobPayment2)
+                .build();
+
+        Job job3 = Job.newBuilder()
+                .setJobStatus(jobStatus)
+                .setJobTitle(jobName)
+                .setJobDescription(jobDescription)
+                .setRequirements(requirements)
+                .setPostExpiry(postExpiry)
+                .setJobDuration(jobDuration)
+                .setLocation(location3)
+                .setJobPay(jobPayment3)
+                .build();
+
+        Job job4 = Job.newBuilder()
+                .setJobStatus(jobStatus)
+                .setJobTitle(jobName)
+                .setJobDescription(jobDescription)
+                .setRequirements(requirements)
+                .setPostExpiry(postExpiry)
+                .setJobDuration(jobDuration)
+                .setLocation(location4)
+                .setJobPay(jobPayment4)
+                .build();
+
+        Job job5 = Job.newBuilder()
+                .setJobStatus(jobStatus)
+                .setJobTitle(jobName)
+                .setJobDescription(jobDescription)
+                .setRequirements(requirements)
+                .setPostExpiry(postExpiry)
+                .setJobDuration(jobDuration)
+                .setLocation(location5)
+                .setJobPay(jobPayment5)
+                .build();
+
+        Job job6 = Job.newBuilder()
+                .setJobStatus(jobStatusExpired)
+                .setJobTitle(jobName)
+                .setJobDescription(jobDescription)
+                .setRequirements(requirements)
+                .setPostExpiry(postExpiry)
+                .setJobDuration(jobDuration)
+                .setLocation(location6)
+                .setJobPay(jobPayment6)
+                .build();
+    
+        // the jobs will be added in a random order
+        firestore.collection(TEST_JOB_COLLECTION).add(job5).get();
+        firestore.collection(TEST_JOB_COLLECTION).add(job1).get();
+        firestore.collection(TEST_JOB_COLLECTION).add(job3).get();
+        firestore.collection(TEST_JOB_COLLECTION).add(job4).get();
+        firestore.collection(TEST_JOB_COLLECTION).add(job2).get();
+        firestore.collection(TEST_JOB_COLLECTION).add(job6).get();
+
+        JobPage expectedJobPage = new JobPage(/* jobList= */ Arrays.asList(job1, job2, job3),
+            /* totalCount= */ 3, Range.between(1, 3));
+
+        // sorting is already defaulted to SALARY and ordering is defaulted to DESCENDING
```

Comment:
nit: `// Act`
^ specially because your Arrange section is really big (probably too big not sure if you can simplify a bit)